src/components/single-line-diagram.js

```diff
@@ -60,11 +61,13 @@ const SvgNotFound = (props) => {
     );
 };
 
-const nosvg = {svg: null, metadata: null, error: null, svgUrl: null};
+const noSvg = {svg: null, metadata: null, error: null, svgUrl: null};
 
 const SingleLineDiagram = (props) => {
 
-    const [svg, setSvg] = useState(nosvg);
+    const [svg, setSvg] = useState(noSvg);
+
+    const studyName = useSelector(state => state.studyName);
```

Comment:
I don't think this component should know the study name. It already gets the full svg url containing the studyname from the parent component. Instead, it should expose a method in its ref (this is done with useImperativeHandle) to refresh the svg